dao/src/main/java/com/epam/esm/dao/repository/impl/UserRepositoryImpl.java

```diff
@@ -42,18 +43,25 @@ public UserRepositoryImpl(EntityManager entityManager) {
   }
 
   @Override
-  public Optional<User> getEntity(Specification<User> specification) {
+  public Optional<Users> getEntity(Specification<Users> specification) {
     try {
       CriteriaBuilder builder = entityManager.getCriteriaBuilder();
-      CriteriaQuery<User> criteriaQuery = specification.getCriteriaQuery(builder);
+      CriteriaQuery<Users> criteriaQuery = specification.getCriteriaQuery(builder);
       return Optional.of(entityManager.createQuery(criteriaQuery).getSingleResult());
     } catch (NoResultException e) {
       return Optional.empty();
     }
   }
 
   @Override
-  public Optional<User> update(User user) {
-    return Optional.of(entityManager.merge(user));
+  public Optional<Users> update(Users users) {
```

Comment:
Don't you need transaction for update method? for save you already have

Reply:
Yes, I suppose you're absolutely right! (fixed)